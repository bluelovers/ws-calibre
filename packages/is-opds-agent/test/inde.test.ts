/**
 * Created by user on 2020/1/27.
 */

///<reference types="jest"/>

import isOPDSAgent, { EnumOPDSAgent, hasFlag } from '../index';

declare module jest
{
	interface Matchers<R, T = {}>
	{
		toHasFlag(expected: number): R
	}
}

expect.extend({
	toHasFlag(received: number, expected: number)
	{
		return {
			message: () =>
				`expected ${received} to has flag ${expected}`,
			pass: Boolean(received & expected),
		};
	},
});

const ua = `OPDS/Stanza iPhone/Aldiko/Moon+ Reader(Android)`;

Object.entries({
	"靜讀天下": {
		ua,
		flags: EnumOPDSAgent.MoonPlusReader,
	},
	"OPDS": {
		ua,
		flags: EnumOPDSAgent.OPDS,
	},
	"Stanza": {
		ua,
		flags: EnumOPDSAgent.Stanza,
	},
	"Aldiko": {
		ua,
		flags: EnumOPDSAgent.Aldiko,
	},
}).forEach(([name, data]) =>
{

	test(`${name}`, async () =>
	{
		let actual = isOPDSAgent(data.ua);
		// @ts-ignore
		expect(actual).toHasFlag(data.flags);
		expect(hasFlag(actual, data.flags)).toBeTruthy();
		expect(actual).toMatchSnapshot();
	})

});

test(`other`, async () =>
{
	let actual = isOPDSAgent(`other`);
	expect(actual).toStrictEqual(0);
});
