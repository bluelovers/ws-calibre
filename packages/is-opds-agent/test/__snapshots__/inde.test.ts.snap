// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aldiko 1`] = `15`;

exports[`OPDS 1`] = `15`;

exports[`Stanza 1`] = `15`;

exports[`靜讀天下 1`] = `15`;
